File: FrontEnd/GranjaSA/src/app/main/Domain/Porcino.ts
import { Cliente } from "./Cliente";
import { RazaPorcino } from "./RazaPorcino";
import { Alimentacion } from "./Alimentacion";

export class Porcino {
    id: string;
    raza: RazaPorcino;
    edad: number;
    peso: number;
    alimentacion: Alimentacion;
    cliente: Cliente | undefined | null;

    constructor(id: string, raza: RazaPorcino, edad: number, peso: number, alimentacion: Alimentacion, cliente: Cliente | undefined | null) {
        this.id = id;
        this.raza = raza;
        this.edad = edad;
        this.peso = peso;
        this.alimentacion = alimentacion;
        this.cliente = cliente;
    }

}

File: BackEnd/src/main/java/com/example/granjasa/Entity/Porcino.java
package com.example.granjasa.Entity;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Entity
@Data
@Table(name = "porcino")
public class Porcino implements Serializable{

    @Id
    private String id;

    @NotNull(message = "La raza es obligatoria")
    @Column(nullable = false)
    private String raza;

    @NotNull(message = "La edad es obligatoria")
    @Column(nullable = false)
    private Integer edad;

    @NotNull(message = "El peso es obligatorio")
    @Column(nullable = false)
    private Integer peso;

    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    @JsonIgnore
    private Cliente cliente;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "alimentacion_id", referencedColumnName = "id")
    private Alimentacion alimentacion;
    
}
